gym/envs/doom/doom_env.py

```diff
@@ -46,22 +121,57 @@ def _render(self, mode='human', close=False):
                 self.viewer = None
             return
         try:
+            if 'human' == mode and self.no_render: return
             state = self.game.get_state()
             img = state.image_buffer
             # VizDoom returns None if the episode is finished, let's make it
             # an empty image so the recorder doesn't stop
             if img is None:
-                img = numpy.zeros((self.screen_height, self.screen_width, 3), dtype=numpy.uint8)
+                img = np.zeros(shape=self.observation_space.shape, dtype=np.uint8)
             if mode == 'rgb_array':
                 return img
             elif mode is 'human':
                 from gym.envs.classic_control import rendering
                 if self.viewer is None:
                     self.viewer = rendering.SimpleImageViewer()
                 self.viewer.imshow(img)
-                sleep(0.02857)  # 35 fps = 0.02857 sleep between frames
+                if 'normal' == self.mode:
+                    sleep(0.02857)  # 35 fps = 0.02857 sleep between frames
         except doom_py.vizdoom.ViZDoomIsNotRunningException:
             pass # Doom has been closed
 
     def _close(self):
         self.game.close()
+
+    def _seed(self, seed=None):
```

Comment:
I've set "nondeterministic=True" in the register() function. We can probably add a #TODO with a warning